### DISCUSSION TITLE
HOWTO control the AC / DC motor using Sonoff 4ch pro.

### DISCUSSION THREAD

**Discussion by TrDA-hab** [arendst/Tasmota#12483] [Show and tell]
### How to use it:  
 - You must add support for Shutter in `my_user_config.h` file (оr flash your ESP8266 module with `tasmota.bin` file).  
 - Run commands in the console to run the "Shutter" mode (you must first configure the GPIO!):  
    `SetOption80 1`     // Enable shutters support.   
    `ShutterMode 1`     // Enable "Shutter mode 1".  
    `Restart 1`         // Restart Tasmota.  
    `Interlock 1,2 3,4` //Group Relay1 and Relay2 in "group 1" and Relay3 and Relay4 in "group 2".  
    `Interlock 1`       //Enable relay interlock .  
    `ShutterRelay1 1`   //Set for relay Relay1 and Relay2.   
    `ShutterRelay2 3`   //Set for relay Relay3 and Relay4.  
    
-  Run commands in the console to test and configure the motor operation: 
   `ShutterOpen1` // Engage the relay to open the shutter1.  
   `ShutterClose1` // Engage the relay to close the shutter1.  
   `ShutterOpenDuration1 5` //Time, in seconds, it takes to fully open the shutter1.  
   `ShutterCloseDuration1 5` // Time, in seconds, it takes to fully close the shutter1.  
   `ShutterOpen2` // Engage the relay to open the shutter2.  
   `ShutterClose2` // Engage the relay to close the shutter2.  
   `ShutterOpenDuration2 8` //Time, in seconds, it takes to fully open the shutter2.  
   `ShutterCloseDuration2 8` // Time, in seconds, it takes to fully close the shutter2.     

![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/Sonoff%204ch%20pro/14-1-1.jpg)  
![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/Sonoff%204ch%20pro/14-1-2.jpg)  
![](https://raw.githubusercontent.com/TrDA-hab/Projects/master/Sonoff%204ch%20pro/14-1-3.jpg)  

**Best regards   
TrDA**

**Comment by namadori** [comment]
I suggest a small change: instead of connecting all the COM pins of the relays to Line, connect:
- Relay1 COM to Line
- Relay2 COM to Relay1 NC
- Relay3 COM to Line
- Relay4 COM to Relay3 NC

This way you have an additional physical interlock, in case of any software failure if both relays are energized the cover will go up and the motor will not break.

**Comment by thundergreen** [comment]
How would be the setup for 2 wire motor?

**Reply by thundergreen** [reply]
got it figured out.. if needed please contact me and i will help out :)

**Reply by orlink** [reply]
Hi, Can you suggest how to setup a 4 wire motor?  One wire needs to switch from N to L while the other from L to N and vise versa in the oposite direction.  The third wire is connected to N permanently and the fourth is switched to L and off. The first two are for the starter coil. The other two for the running coil.

**Reply by Phil636** [reply]
@thundergreen I am searching for a 2 wire scheme. Can you help me? 😀

**Reply by imaxi96** [reply]
@thundergreen Can you maybe share your 2-wire scheme with me? :) Thanks

**Reply by skerchner** [reply]
Can you share the 2 wire schematic with me?  Thanks!
